library/ssl_tls13_server.c

```diff
@@ -1042,8 +1044,11 @@ int ssl_parse_client_psk_identity_ext( mbedtls_ssl_context *ssl,
                                   truncated_clienthello_start, truncated_clienthello_end - truncated_clienthello_start, server_computed_binder );
         }
         else {
-            /* Case 2: We are using an externally configured PSK */
-            ret = ssl_calc_binder( ssl, ssl->conf->psk, ssl->conf->psk_len,
+            /* Case 2: We are using a static PSK, or a dynamic PSK if one is defined */
+            if( mbedtls_ssl_get_psk( ssl, &psk, &psk_len ) == MBEDTLS_ERR_SSL_PRIVATE_KEY_REQUIRED )
```

Comment:
Minor: At the moment, `mbedtls_ssl_get_psk()` can only return `MBEDTLS_ERR_SSL_PRIVATE_KEY_REQUIRED`, but that might change. In this case, you don't want to continue if another error has been returned. Therefore, I'd suggest to simply safe the return value in `ret`, and return immediately if it's non-zero.

Reply:
```suggestion
            ret = mbedtls_ssl_get_psk( ssl, &psk, &psk_len );
            if( ret != 0 )
                return( ret );
```